synapse/storage/util/id_generators.py

```diff
@@ -570,6 +574,19 @@ def _update_stream_positions_table_txn(self, txn):
         pos = (self.get_current_token_for_writer(self._instance_name),)
         txn.execute(sql, (self._stream_name, self._instance_name, pos))
 
+    def _update_stream_positions_table_conn(self, conn: LoggingDatabaseConnection):
+        # We use autocommit here so that we don't have to go through a
+        # transaction dance, which a) adds latency and b) runs the risk of
+        # serialization errors.
```

Comment:
don't we still risk serialization errors, but now they won't be handled?

autocommit is just syntactic sugar for a transaction with explicit commit afaik?

Reply:
I believe it'll behave (at worst) like it was in a transaction with a `READ COMMITTED` isolation level, which doesn't have `concurrent update` errors. (Plus the fact that the command gets committed immediately means the likelihood of concurrent updates is dramatically smaller.


> ... but now they won't be handled?

Ah, I forgot that we handled those exceptions in the `new_transaction` function, rather than in `runWithConnection`. I do think that we'll still get errors even if we're not in a transaction though?

Reply:
Of course having said all that the CI is failing with:

```
psycopg2.errors.SerializationFailure: could not serialize access due to concurrent update
```